### PR TITLE
nrf: move from /drivers to /libs

### PR DESCRIPTION
**Pull request by lemrey**
This PR moves several libraries from `/drivers` to `/lib` based on the criteria that to qualify as a driver, a piece of code must ~~implement `device.h`~~ follow the Zephyr device driver model. A couple of these should probably be reworked a little to implement `device.h`, since they really are device drivers after all.

### CODE REVIEW

**Comment by lemrey** [comment]
@ru-fu The doc is failing to build because:
```
/jenkins_workspace/workspace/connect-nrf_documentation_master/_build/nrf/rst/doc/nrf/drivers.rst:10: WARNING: toctree glob pattern '../../include/drivers/*' didn't match any documents
```

The path is now empty and so it can't be tracked in `git`.
Shall we have a placeholder file of some kind there, so that we can keep the path both for the doc and for the `CODEOWNER` file in the future?

**Comment by rlubos** [comment]
I think this PR needs more reviewers. 

What does it mean that the piece of code does not implement `device.h`? Given the example of `lte_lc.c`, it does use device macros to register the device, ain't that enough? I stick to this module, as it was left under the `drivers` subdirectory only because we have a very similar module, `drivers/modem` lying exactly in there, we're just following the upstream convention (though I find the naming used upstream to fit better the "device" convention).

Anyway, I have no issue with moving stuff around if it's justified.

**Comment by lemrey** [comment]
For reference: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1608.
To clarify, I think that code under `/drivers` should follow patterns like other device drivers (GPIO, FLASH etc) e.g. `device_get_binding()` and passing the `struct device` around as a context.

File: include/st25r3911b_nfc.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ API documentation
 *****************
 
 | Header file: :file:`include/st25r3911b_nfca.h`
```
Comment (ru-fu) [inline include/st25r3911b_nfc.rst]: Should this file go to include/nfc/...?

**Comment by lemrey** [comment]
I would say that `drivers` are implementations that follow the Zephyr device driver model, which is documented here: https://docs.zephyrproject.org/latest/reference/drivers/index.html

Notably, the device driver model is based on the `struct device` abstraction.

**Comment by SebastianBoe** [comment]
There is no technical difference between the drivers folder and the lib folder. The separation exists only to make it easier for users to find what they are looking for.

Therefore which directory something goes in should be based on where the user thinks it will be. Perhaps this should be based on whether it follows the Zephyr device driver model, or perhaps it should be based on an intuitive guess for whether something is a driver or not.

This is not clear to me.

**Comment by anangl** [comment]
I am not convinced about moving `gps_sim` and `nrf9160_gps` to `drivers/sensor/`. Such placement may turn out misleading. These two in fact do not implement the API defined in `sensor.h`. The API defined in `gps.h` is only similar to that one in some points.

**Comment by jtguggedal** [comment]
I was just about to comment the same as @anangl, I think the GPS drivers should probably not be within `drivers/sensor/`, as they don't use the sensor API.

Also, tend to agree with @SebastianBoe's point that people would expect to find something like the driver for `ADP536x` in `drivers/` and not `lib/`. I don't have very strong opinions on this, though, so if there's consensus to separate stuff based on whether it implements `device` model or not, just go ahead.

**Comment by lemrey** [comment]
I agree, `nrf9160_gps` and `gps_sim` can be left in `/drivers`, I was tricked by some API names that match those defined for `sensor.h` :)

I see @SebastianBoe point, but upstream drivers follow a specific model while most of the "drivers" we have downstream don't. It's true that customers expect to find something like `ADP356` under `/drivers`, but I argue that `ADP356` should follow the same APIs as drivers upstream, and not reinvent it; we will end up having a bunch of different APIs for each driver otherwise.

**Comment by hakonfam** [comment]
[CHANGELOG POLICE] -> Please update changelog

**Comment by ru-fu** [comment]
I added a commit to update the text for the drivers overview page.

**Comment by lemrey** [comment]
Moved the `nrf9160_gps` and `gps_sim` to `/drivers/gps` and their include `gps.h` under `/include/drivers`.

**Comment by lemrey** [comment]
@anangl @SebastianBoe please revisit.

**Comment by SebastianBoe** [comment]
If this change is OK with @anangl , it is OK with me.

File: drivers/gps/gps_sim/gps_sim.c [hunk 2]
```diff
@@ -0,0 +1,393 @@
+/*
```
Comment (anangl) [inline drivers/gps/gps_sim/gps_sim.c]: This file was supposed to be moved not copied. Same goes for `nrf9160_gps.c`.

Comment (lemrey) [reply drivers/gps/gps_sim/gps_sim.c]: Thanks, that was a mistake

**Comment by lemrey** [comment]
Updated changelog.

**Pull request merged**